utils/src/main/scala/za/co/absa/enceladus/utils/testUtils/LoggerTestBase.scala

```diff
@@ -19,18 +19,41 @@ import java.io.ByteArrayOutputStream
 
 import org.apache.spark.sql.DataFrame
 import org.slf4j.{Logger, LoggerFactory}
+import LogLevel._
 
 trait LoggerTestBase {
+
   val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
-  protected def logDataFrameContent(df: DataFrame): Unit = {
-    logger.debug(df.schema.treeString)
+  def logLevelToLogFunction(logLevel: LogLevel): String => Unit = {
+    logLevel match {
+      case Trace => logger.trace
+      case Debug => logger.debug
+      case Info  => logger.info
+      case Warn  => logger.warn
+      case Error => logger.error
+    }
+  }
+
+  protected def logDataFrameContent(df: DataFrame, logLevel: LogLevel = Debug): Unit = {
+    val logFnc = logLevelToLogFunction(logLevel)
+    logFnc(df.schema.treeString)
 
     val outCapture = new ByteArrayOutputStream
     Console.withOut(outCapture) {
       df.show(truncate = false)
     }
     val dfData = new String(outCapture.toByteArray).replace("\r\n", "\n")
-    logger.debug(dfData)
+    logFnc(dfData)
   }
 }
+
+sealed trait LogLevel
+
+object LogLevel {
```

Comment:
Is there a reason for not using [slf4j's log level enum](https://www.slf4j.org/apidocs/org/slf4j/event/Level.html)

Reply:
Now using it, Thanks for bringing this to my knowledge.